ui: Add per-node option to Custom Graphs

mrtracy opened the pull request:
Adds "Breakdown" aggregator option to custom graphs, which will graph a
separate line for every node in the cluster for a given metric.

Also fixes a bug where you could not graph multiple lines of the same
named metric.

Release note: None

<img width="1335" alt="screen shot 2018-03-12 at 9 48 21 pm" src="https://user-images.githubusercontent.com/6969858/37318139-2de59d6e-263f-11e8-8a79-55a3795c8def.png">

Commit:
ui: Add per-node option to Custom Graphs

Adds "Per Node" option to custom graphs, which will graph a separate
line for every node in the cluster for a given metric.

Also fixes a bug where you could not graph multiple lines of the same
named metric.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23771)
<!-- Reviewable:end -->

vilterp commented:
This is cool — tried locally and it works good 👍 Starting to remind me of [honeycomb.io](http://honeycomb.io/)'s query editor:
![image](https://user-images.githubusercontent.com/7341/37425660-82b24ba6-279a-11e8-89f6-2cb1505f7522.png)

The "breakdown" value of 999 seems a little weird; maybe we can wrap aggregation/breakdown in another type? Any thoughts @couchand?

Aside: reduces the need for #23733… Maybe we should link from the current graphs to "broken-down" graphs here.

vilterp commented:
Capturing in-person discussion: would prefer a separate column with a checkbox that says "per-node" (or something), instead of "breakdown" as an aggregation option with a sentinel value.

mrtracy commented:
Updated with suggestion (also updated commit message).

![screen shot 2018-03-14 at 5 53 58 pm](https://user-images.githubusercontent.com/6969858/37433085-f243aa80-27b0-11e8-83c0-cd5924d74762.png)

vilterp reviewed (approved):
Love it. LGTM pending lint fixes.
